app/models/spree/image.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Spree
+  class Image < Asset
+    validates_attachment_presence :attachment
+    validate :no_attachment_errors
+
+    has_attached_file :attachment,
+                      styles: { mini: '48x48>', small: '100x100>',
+                                product: '240x240>', large: '600x600>' },
+                      default_style: :product,
+                      url: '/spree/products/:id/:style/:basename.:extension',
+                      path: ':rails_root/public/spree/products/:id/:style/:basename.:extension',
+                      convert_options: { all: '-strip -auto-orient -colorspace RGB' }
+
+    # save the w,h of the original image (from which others can be calculated)
+    # we need to look at the write-queue for images which have not been saved yet
+    after_post_process :find_dimensions
+
+    include Spree::Core::S3Support
+    supports_s3 :attachment
+
+    Spree::Image.attachment_definitions[:attachment][:styles] =
+      ActiveSupport::JSON.decode(Spree::Config[:attachment_styles]).symbolize_keys!
+    Spree::Image.attachment_definitions[:attachment][:path] = Spree::Config[:attachment_path]
+    Spree::Image.attachment_definitions[:attachment][:url] = Spree::Config[:attachment_url]
+    Spree::Image.attachment_definitions[:attachment][:default_url] =
+      Spree::Config[:attachment_default_url]
+    Spree::Image.attachment_definitions[:attachment][:default_style] =
+      Spree::Config[:attachment_default_style]
```

Comment:
:clap: :clap: now we can finally DRY ImageSettingsController#update_paperclip_settigns!

Reply:
image settings controller will die here: https://github.com/openfoodfoundation/openfoodnetwork/pull/5729/files